Objective-C/CBLDatabase.mm

```diff
@@ -259,18 +248,13 @@ - (BOOL) inBatch: (NSError**)outError do: (void (^)())block {
 }
 
 
-- (CBLDocument*) document {
-    return [self documentWithID: [self generateDocID]];
+- (CBLDocument*) documentWithID: (NSString*)documentID {
```

Comment:
It's correct to use mustExist = YES. 

How about new doc with given ID.
> It will return null. The new API spec doesn't do get or create anymore. This method is only get.